### PR TITLE
[Task-1]  Setup the project using Rails Template

### PR DESCRIPTION
**Pull request by mcneto**
Task - https://github.com/mcneto/ruby-google-search-ic/issues/1

## What happened 👀

- Initialize the project using Rails Template
- Update README file (Introduction, About)
 
## Insight 📝

I will setup and config the testing part later in #2, #3 
 
## Proof Of Work 📹

Screen Shot 2564-06-11 at 13 32 17
Screen Shot 2564-06-11 at 18 56 27
Screen Shot 2564-06-11 at 18 55 46

### CODE REVIEW

File: .rspec [hunk 1]
```diff
@@ -0,0 +1,3 @@
+--color
+--require spec_helper
+--format documentation
```
Comment (junan) [inline .rspec:3]: Please add a new line end of the file
```suggestion
--format documentation

```

File: Procfile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+web: bin/rails server -p $PORT -e $RAILS_ENV
+worker: bundle exec sidekiq -C config/sidekiq.yml
```
Comment (junan) [inline Procfile:2]: Same as previous comment

File: Procfile.dev [hunk 3]
```diff
@@ -0,0 +1,3 @@
+web: rails s -p 3000 -b '0.0.0.0'
+worker: bundle exec sidekiq -C config/sidekiq.yml
+webpack: yarn && bin/webpack-dev-server
```
Comment (junan) [inline Procfile.dev:3]: Same as previous comment

File: config/application.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    config.load_defaults 6.1
+
+    # Configuration for the application, engines, and railties goes here.
+    #
+    # These settings can be overridden in specific environments using the files
+    # in config/environments, which are processed later.
+    #
+    # config.time_zone = "Central Time (US & Canada)"
+    # config.eager_load_paths << Rails.root.join("extras")
+
```
[the remark this answers is not in the document]
Comment (junan) [reply config/application.rb:30]: @mcneto Lets remove the trailing whitespace
```suggestion

```

File: config/environments/development.rb [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  # Don't care if the mailer can't send.
+  config.action_mailer.raise_delivery_errors = false
+
+  config.action_mailer.perform_caching = false
+
+  config.action_mailer.delivery_method = :letter_opener
+
+  config.action_mailer.asset_host = ENV.fetch('MAILER_DEFAULT_HOST')
+
+  config.action_mailer.default_url_options = {
```
Comment (junan) [inline config/environments/development.rb:45]:
```suggestion
  config.action_mailer.default_url_options = {
```

File: config/environments/production.rb [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+  # Use a real queuing backend for Active Job (and separate queues per environment).
+  # config.active_job.queue_adapter     = :resque
+  # config.active_job.queue_name_prefix = "google_search_ic_production"
+
+  config.action_mailer.perform_caching = false
+
+  config.action_mailer.asset_host = ENV.fetch('MAILER_DEFAULT_HOST')
+
+  config.action_mailer.default_url_options = {
```
Comment (junan) [inline config/environments/production.rb:69]:
```suggestion
  config.action_mailer.default_url_options = {
```

File: config/environments/test.rb [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+  # Disable request forgery protection in test environment.
+  config.action_controller.allow_forgery_protection = false
+
+  # Store uploaded files on the local file system in a temporary directory.
+  config.active_storage.service = :test
+
+  config.action_mailer.perform_caching = false
+
+  config.action_mailer.default_url_options = {
```
Comment (junan) [inline config/environments/test.rb:43]:
```suggestion
  config.action_mailer.default_url_options = {
```

File: config/initializers/backtrace_silencers.rb [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+# Add backtrace silencer
+# Default silencer is removing backtrace involving engines files
+# Our silencer also checks for engines directory
+# Ref: https://github.com/rails/rails/blob/master/railties/lib/rails/backtrace_cleaner.rb
+Rails.backtrace_cleaner.add_silencer do |line|
+  !(Rails::BacktraceCleaner::APP_DIRS_PATTERN.match?(line) || /^engines/.match?(line))
+end
+
+
```
Comment (junan) [inline config/initializers/backtrace_silencers.rb:18]:
```suggestion
```

File: config/initializers/backtrace_silencers.rb [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+Rails.backtrace_cleaner.remove_silencers!
+
+# Add backtrace silencer
+# Default silencer is removing backtrace involving engines files
+# Our silencer also checks for engines directory
+# Ref: https://github.com/rails/rails/blob/master/railties/lib/rails/backtrace_cleaner.rb
+Rails.backtrace_cleaner.add_silencer do |line|
+  !(Rails::BacktraceCleaner::APP_DIRS_PATTERN.match?(line) || /^engines/.match?(line))
+end
+
```
Comment (junan) [inline config/initializers/backtrace_silencers.rb:17]: Please remove extra two lines

File: config/environments/development.rb [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    host: ENV.fetch('MAILER_DEFAULT_HOST'),
```
[the remark this answers is not in the document]
Comment (junan) [reply config/environments/development.rb:46]: @mcneto Please remove the trailing whitespace from here( also check other bot comments in `production.rb` and `test.rb` file)

File: config/environments/test.rb [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+  # Tell Active Support which deprecation messages to disallow.
+  config.active_support.disallowed_deprecation_warnings = []
+
+  # Raises error for missing translations.
+  # config.i18n.raise_on_missing_translations = true
+
+  # Annotate rendered view with file names.
+  # config.action_view.annotate_rendered_view_with_filenames = true
+
```
[the remark this answers is not in the document]
Comment (junan) [reply config/environments/test.rb:67]: @mcneto There are more trailing whitespace to remove

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 12]
```diff
@@ -1,13 +1,13 @@
-https://github.com/nimblehq/git-template/issues/??
+Add the story URL here. Prefer the short link format, e.g. https://app.clubhouse.io/acme/story/1234/
```
Comment (rafayet-monon) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: As we will be using github project here, you can add Task card here and add respective card link in each PR
```suggestion
Task card - 
```

File: app/assets/stylesheets/vendor/bootstrap/bootstrap.scss [hunk 13]
```diff
@@ -0,0 +1,42 @@
+// By default every component is imported
```
Comment (rafayet-monon) [inline app/assets/stylesheets/vendor/bootstrap/bootstrap.scss:1]: As per this comment you need to pick only the imports you need and comment out the rest. For example, we can start with the below essential components for now and comment out the rest
image

File: config/application.rb [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+    # Prefix the queue name of all jobs with Rails ENV
+    config.active_job.queue_name_prefix = Rails.env
+
+    # Compress the responses to reduce the size of html/json controller responses.
+    config.middleware.use Rack::Deflater
+
+    # Automatically generate the `translation.js` files
+    config.middleware.use I18n::JS::Middleware
+end
```
Comment (rafayet-monon) [inline config/application.rb:46]: indentation doesn't seem right here 🤔

**Review by rafayet-monon — COMMENTED** [review]
Let's fix the CI errors 💪

File: .semaphore/README.md [hunk 15]
```diff
@@ -0,0 +1,49 @@
+# Semaphore CI 2.0
```
Comment (junan) [inline .semaphore/README.md:1]: We don't need semaphore files as we will use Github Actions for CI/CD. Yu can remove whole `.semaphore` directory 👍

File: Dockerfile [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+      apt-get install -y --no-install-recommends google-chrome-stable && \
+      rm /etc/apt/sources.list.d/google-chrome.list && \
+      apt-get clean && \
+      rm -rf /var/lib/apt/lists/* ; \
+    fi
+
+WORKDIR $APP_HOME
+
+# Nginx config
+COPY config/nginx/app.conf.template /etc/nginx/conf.d/default.conf
```
Comment (junan) [inline Dockerfile:56]: If you dont  have a plan to use NginX with Heroku. Please remove it in your CD PR (Note: I personally did not use NginX with Heroku in my ICs)

Comment (junan) [inline Dockerfile:56]: If you dont have a plan to use `nginx ` you can remove it in your CD PR. There is some `nginx` reference in another place also, please remove them also incase you dont use nginx with Heroku. You can check `malparty`'s repo also for reference https://github.com/malparty/google-search-ruby/

**Review by junan — APPROVED** [review]
I approved with a comment

File: Dockerfile (hunk 16 above)
Comment (mcneto) [reply Dockerfile:56]: Okay, I'll check this. Thank you 🙂

**Review by rafayet-monon — APPROVED** [review]

**Pull request merged**